File: sdks/csharp/HookSniff/BackgroundTask.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HookSniff
{
    public class BackgroundTask
    {
        private readonly HookSniffClient _client;

        public BackgroundTask(HookSniffClient client)
        {
            _client = client;
        }

        public Task<List<Dictionary<string, object>>> List()
        {
            return _client.HookSniffHttpClient.GetAsync<List<Dictionary<string, object>>>("/api/v1/background-tasks");
        }

        public Task<Dictionary<string, object>> Get(string taskId)
        {
            return _client.HookSniffHttpClient.GetAsync<Dictionary<string, object>>($"/api/v1/background-tasks/{taskId}");
        }

        public Task<Dictionary<string, object>> Cancel(string taskId)
        {
            return _client.HookSniffHttpClient.PutAsync<Dictionary<string, object>>($"/api/v1/background-tasks/{taskId}", null);
        }
    }
}

File: sdks/ruby/lib/hooksniff/background_task.rb
# frozen_string_literal: true

module HookSniff
  class BackgroundTask
    def initialize(api_client)
      @api_client = api_client
    end

    def list
      @api_client.execute(method: :get, path: "/api/v1/background-tasks")
    end

    def get(task_id)
      @api_client.execute(method: :get, path: "/api/v1/background-tasks/#{task_id}")
    end

    def cancel(task_id)
      @api_client.execute(method: :put, path: "/api/v1/background-tasks/#{task_id}")
    end
  end
end
